src/Inkluzitron/Handlers/InviteHandler.cs

```diff
@@ -0,0 +1,74 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Discord.WebSocket;
+using Inkluzitron.Data;
+using Inkluzitron.Extensions;
+using Inkluzitron.Services;
+using Microsoft.EntityFrameworkCore;
+
+namespace Inkluzitron.Handlers
+{
+    /// <summary>
+    /// Handler to catch event on joining new user
+    /// </summary>
+    public class InviteHandler : IHandler
+    {
+        private DiscordSocketClient DiscordSocketClient { get; }
+
+        private DatabaseFactory DatabaseFactory { get; }
+        private UsersService UsersService { get; }
+
+        public InviteHandler(DiscordSocketClient discordSocketClient,
+            DatabaseFactory databaseFactory,
+            UsersService usersService)
+        {
+            DiscordSocketClient = discordSocketClient;
+            DatabaseFactory = databaseFactory;
+            UsersService = usersService;
+
+            DiscordSocketClient.UserJoined += OnUserJoinedAsync;
+        }
+
+        /// <summary>
+        /// Checks which invite link has been used by newly joined user and
+        /// updates corresponding record in database for it
+        /// </summary>
+        /// <param name="user">newly joined user</param>
+        private async Task OnUserJoinedAsync(SocketGuildUser user)
+        {
+            var dbContext = DatabaseFactory.Create();
```

Comment:
```suggestion
            using var dbContext = DatabaseFactory.Create();
```

Pokud nepoužiješ using a navoláš ručně někde dispose, tak ti tam budou smrdět memory leaky z EF.

Reply:
Puvodne jsem to tam mel, ale delalo to nejaky problemy a pak to zapomnel vyzkouset po dokonceni. Chyba samozrejme byla nekde jinde, takze to tam vratim.